acnportal/acnsim/events/event.py

```diff
@@ -154,19 +178,48 @@ def _from_dict(
         loaded_dict: Optional[Dict[str, BaseSimObj]] = None,
     ) -> Tuple[BaseSimObj, Dict[str, BaseSimObj]]:
         """ Implements BaseSimObj._from_dict. """
-        out_obj = cls(
-            attribute_dict["timestamp"],
-            attribute_dict["station_id"],
-            attribute_dict["session_id"],
-        )
+        # noinspection PyProtectedMember
+        try:
+            ev, loaded_dict = BaseSimObj._build_from_id(
+                attribute_dict["ev"], context_dict, loaded_dict=loaded_dict
+            )
+        except KeyError:
```

Comment:
@zach401 deserialization logic for loading serialized instances of the old UnplugEvent implementation. The encapsulated EV object is missing attributes; in theory, it's possible to build the full UnplugEvent object (with a fully attributed EV) using the ev_history attribute of the Simulator, but this would need to be done after loading. We could provide a utility to do this in a separate PR, should be simple.

Reply:
Yeah, that might be interesting to add in a future PR. What do you think about using None for the missing attributes instead of -1? Is deleting the attributes the best course of action. I think it's fine for now, we can readdress this in the PR that pulls from the ev history. It's not a big deal since this will be a little-used feature most likely.